src/spineml/simulator/inputValue.cc

```diff
@@ -323,11 +303,39 @@ void SpineMLSimulator::InputValue::External::update(double, unsigned int,
     }
 }
 
+//----------------------------------------------------------------------------
+// SpineMLSimulator::InputValue::ExternalNetwork
+//----------------------------------------------------------------------------
+SpineMLSimulator::InputValue::ExternalNetwork::ExternalNetwork(double dt, unsigned int numNeurons, const pugi::xml_node &node)
+: External(dt, numNeurons, node)
+{
+    // Read connection stats
+    const std::string connectionName = node.attribute("name").value();
+    const std::string hostname = node.attribute("host").value();
+    const int port = node.attribute("tcp_port").as_int();
```

Comment:
Out of interest, does pugiXML throw an exception if these attributes are not present? Or should we be testing for this?

Reply:
It returns empty strings (https://pugixml.org/docs/manual.html#access.attrdata) so the failure mode should be that the network client barfs

Reply:
Ah the good ol' ``std::barf_exception``!